packages/core/lib/use-puck-history.ts

```diff
@@ -0,0 +1,71 @@
+import type { Data } from "../types/Config";
+import { useEffect } from "react";
+import { PuckAction } from "./reducer";
+import { useActionHistory } from "./use-action-history";
+import { useHotkeys } from "react-hotkeys-hook";
+import EventEmitter from "event-emitter";
+import { useDebouncedCallback, useDebounce } from "use-debounce";
+import { diff, applyChange, revertChange } from "deep-diff";
+
+const DEBOUNCE_TIME = 100;
+const RECORD_DIFF = "RECORD_DIFF";
+export const historyEmitter = EventEmitter();
+export const recordDiff = (newData: Data) =>
+  historyEmitter.emit(RECORD_DIFF, newData);
+
+export function usePuckHistory({
+  data,
+  dispatch,
+}: {
+  data: Data;
+  dispatch: (action: PuckAction) => void;
+}) {
+  const { canForward, canRewind, rewind, forward, record } = useActionHistory();
+
+  useHotkeys("meta+z", rewind, { preventDefault: true });
+  useHotkeys("meta+y", forward, { preventDefault: true });
```

Comment:
Sorry, on macOS redo is actually CMD + SHIFT + Z.

Perhaps we should split this into two hotkeys?

```suggestion
  useHotkeys("cmd+shift+z", forward, { preventDefault: true });
  useHotkeys("ctrl+y", forward, { preventDefault: true });
```

or just define meta for both - I don't think it hurts

```suggestion
  useHotkeys("meta+shift+z", forward, { preventDefault: true });
  useHotkeys("meta+y", forward, { preventDefault: true });
```

(untested examples)

Reply:
cool! I like second one! (I checked it work)